app/lib/account/consent_backend.dart

```diff
@@ -187,14 +178,18 @@ class ConsentBackend {
     final c = await _db.lookupValue<Consent>(
         _db.emptyKey.append(Consent, id: consentId),
         orElse: () => null);
-    if (c == null) return null;
+    if (c == null) {
+      throw NotFoundException.resource('consent: $consentId');
+    }
 
     // Checking that consent is for the current user.
-    if (c.userIdOfConsent != null && c.userIdOfConsent != user.userId) {
-      return null;
+    if (c.userIdOfConsent != null) {
+      InvalidInputException.check(c.userIdOfConsent == user.userId,
+          'Consent.userId does not match current User.');
     }
-    if (c.email != null && c.email != user.email) {
-      return null;
+    if (c.email != null) {
+      InvalidInputException.check(
+          c.email == user.email, 'Consent.email does not match current User.');
```

Comment:
What about:
`'This invitation is not for the user account currently logged in'`